libipl/p10/ipl0.C

```diff
@@ -193,12 +195,14 @@ static int update_hwas_state_callback(struct pdbg_target* target, void *priv)
 
 	} else if (ipl_type() == IPL_TYPE_NORMAL) {
 
-		if (type == TGT_TYPE_PROC) {
-			if (ipl_is_master_proc(target)) {
-				ipl_log(IPL_INFO,
-					"Primary processor [%s] is guarded "
-					"so, skipping to apply", tgtPhysDevPath);
-				return GUARD_PRIMARY_PROC_NOT_APPLIED;
+		if (!openpower::guard::isEphemeralType(target_info->guardType)) {
```

Comment:
This  is aprtially fixing the issue, still need to look , graceful reboot follwed by primary processor guard use case.

Reply:
and also, I think we need to see a graceful reboot followed by the primary processor not functional without a guard record

IMO, we need to see some better way to handle the primary processor in the **updatehwmodel** irrespective of the guard record.

Do we need to assume or mark the primary processor as functional if it is not functional in the istep0 and proceeds further, let's fail in the reboot attempt and shut down the host if the primary is really bad?

I guess istep0 doesn't guarantee the primary processor is bad or not except through hardware procedures failure, right?

Reply:
Need bigger discussion on Primary processor deconfig record handling , will track that support separately